src/mrt.rs

```diff
@@ -243,6 +250,14 @@ impl TryFrom<u32> for Ticks {
     }
 }
 
+// Note that we don't implement `From` (which would imply `Into` as well) since `From` might fail
+// (`u32` value too big) but `Into` won't (if it fits into `Ticks`, it fits into an `u32`)
+impl Into<u32> for Ticks {
```

Comment:
Generally, implementing `From` should be preferred to implementing `Into`, as the core library provides a blanket `Into` impl for all `From`s.

`impl From<Ticks> for u32` should be possible in this case, without the compiler complaining, but I might be wrong.

Reply:
From what I've gathered `From` is not allowed to fail, and from what I gather the conversion will fail if you pass `0_xff_fff_fff` / `u32::MAX` (see https://github.com/lpc-rs/lpc8xx-hal/blob/85923c568c07298e3b2b17bbbe81dd13f7f7837c/src/mrt.rs#L241), which is why I picked this route. Did I miss a better way to resolve this?

Reply:
The conversion from `Ticks` to `u32` (which is what you've implemented here) is infallible. I'm just suggesting another way to formulate the same conversion:
``` rust
impl From<Ticks> for u32 {
    // ...
}
```

Quoting from the [documentation for `Into`](https://doc.rust-lang.org/std/convert/trait.Into.html):
> One should avoid implementing `Into` and implement `From` instead. Implementing `From` automatically provides one with an implementation of `Into` thanks to the blanket implementation in the standard library.

So implementing `Into` should only be done, if `From` can't be implemented for some reason. I think my suggestion above should work though, because `From<Ticks>` is "local enough" (I know there's a better way to say this, but I'm not familiar with all the Rust lingo).

Reply:
Ugh! Yeah you're right– I was thinking about it the wrong way round. Will open a PR together with mitigation for your other comment this evening.